src/Driver/OCI8/Exception/Error.php

```diff
@@ -24,6 +28,18 @@ public static function new($resource): self
         $error = oci_error($resource);
         assert($error !== false);
 
-        return new self($error['message'], null, $error['code']);
+        $code    = $error['code'];
+        $message = $error['message'];
+        if ($code === self::CODE_TRANSACTION_ROLLED_BACK) {
+            // There's no way this can be unit-tested as it's impossible to mock $resource
+            //ORA-02091: transaction rolled back
+            //ORA-00001: unique constraint (DOCTRINE.GH3423_UNIQUE) violated
+            [$firstMessage, $secondMessage] = explode("\n", $message, 2);
+
+            [$code, $message] = explode(': ', $secondMessage, 2);
+            $code             = (int) str_replace('ORA-', '', $code);
```

Comment:
The unique constraint violation error is of interest to us in this specific case but there may be other cases. I don't think that replacing the error reported by the driver based on a personal preference is a good idea.

Instead, such errors could be represented as a chain (see `Exception::$previous`) of exceptions. So it would be 1) a transaction exception caused by 2) a unique constraint violation.

Reply:
I have to introduce custom TransactionRolledBack exception

Reply:
How do I make it work with ExceptionConverter then? It does not support chained errors from the driver.

This is what I have since I'm able to read multiple errors from the server now:

`Error - Transaction rolled back <- Error - UC violation`

--------------

This is what exception converter does right now

`Driver Exception (Error - Transaction rolled back <- Error - UC violation)`

---------------

We need to somehow inject `UniqueConstraintViolationException` into 👆🏾 but cannot simply use `previous` as OCI Error is used as `previous` now.

Reply:
> How do I make it work with ExceptionConverter then? It does not support chained errors from the driver.

That problem sounds solvable to me.

> Instead, such errors could be represented as a chain (see `Exception::$previous`) of exceptions. So it would be 1) a transaction exception caused by 2) a unique constraint violation.

This is the way.